### DISCUSSION TITLE
Ability to send a private message directly on github.com

### DISCUSSION THREAD

**Discussion by prettysync** [community/community#15580] [Other Feature Feedback, Questions, & Ideas]
It would be nice if we could send private messages to users directly on github.com. With the ability to only allow followers to contact us.

**Comment by ghost** [comment]
Hi! @wealthysync 

1. Here's a ui/ux idea based on your idea... 
2. I thought the same thing... about sending messages private in GitHub. 
3. Hope to help GitHub and the GitHub community

#### ui/ux
![image](https://user-images.githubusercontent.com/95204402/168397597-a17d844d-cc9f-4f34-9d1e-d9e55c7906a2.png)

**Reply by ghost** [reply]
Hi all!

1. I have this idea to help in this topic... It's just a suggestion I thought: "to avoid spam or bad practices, it would be interesting to have a limit of 1,000 messages per day, I thought of this number of 1,000 messages per day because on Twitter there is this limit" - Direct Messages (daily): The limit is 1,000 messages sent per day. reference here [help.twitter.com/en/rules-and-policies](https://help.twitter.com/en/rules-and-policies/twitter-limits)
2. I would like to have some feedback, what do you guys think of the idea?

**Reply by prettysync** [reply]
Hello! Your idea for the limit of messages to send is interesting. 
**It can help regulate traffic on the GitHub servers, and limit unwanted messages.**

**Reply by upintheairsheep** [reply]
reduce it to 500 or 250?

**Reply by SiddhantTotade** [reply]
:heart:

**Reply by peterkulko** [reply]
It`s good idea 👍

**Reply by darshk07** [reply]
reduce it to 100?

**Comment by prettysync** [comment]
This could allow above all to be on all social networks: as a company, your presence on social networks conveys:
- Your brand name,
- Your benefits and services,
- Your corporate image (friendly, premium).

Allowing messages to be sent to GitHub accounts would remove the social barrier between companies and developers.

**Reply by nicoleCodeGirl** [reply]
Yes, I want to be able to message another GitHub user. But, because of spamming, only recieve messeges from people we follow.

**Comment by ghost** [comment]
Hi everyone!

#### concept
![image](https://user-images.githubusercontent.com/108383570/176567704-203c1619-910c-417f-a919-7431ce740b18.png)

#### image description
this image is an idea of this feature

**Reply by armanmoztar** [reply]
I really like this idea!

**Reply by ghost** [reply]
@armanmoztar thank you for feedback ;D - I really liked your comment

**Reply by mishmanners** [reply]
A lot of people have asked about private DMs, however there are many ways to contact people via GitHub. Those who want to be contacted add their Twitter/X links to their profile, and many put social media links and email contact details in their READMEs. If you can't find a way to contact someone, they usually don't want to be contacted. There are a lot of sites where DMs have been allowed, and it has devolved into a mess of criticism and hate speech towards users. I understand the potential benefit of having DMs, but I also think the downsides are much great. [See my GitHub profile as an example of someone who wants to/is easily contactable](https://github.com/mishmanners).

**Comment by melvincwng** [comment]
Great idea! +1 👍🏻

**Comment by ghost** [comment]
Hi all! The cool thing would be to be able to set a time when the message will be deleted or have the option to delete the private message. Here is an interesting topic that could help this topic here: https://github.com/community/community/discussions/16739

**Comment by jfullstackdev** [comment]
I answered here, so I'm sharing it 

https://github.com/orgs/community/discussions/17888

**Reply by ghost** [reply]
@jdevstatic true!

**Comment by MeesJ** [comment]
This would be great and very helpful for collaboration with other developers. GitHub should definitely consider adding this feature.

**Reply by prettysync** [reply]
agree!

**Reply by upintheairsheep** [reply]
me too! user issues would also be nice

**Reply by upintheairsheep** [reply]
I am sometimes forced to write unrelated issues on projects or unrelated comments on issues to get in touch. Most workplaces block social media platforms to prevent distractions so this would be helpful. Right now, I try to find the code of their github site or user info page and issue there.

**Reply by prettysync** [reply]
> I am sometimes forced to write unrelated issues on projects or unrelated comments on issues to get in touch. Most workplaces block social media platforms to prevent distractions so this would be helpful. Right now, I try to find the code of their github site or user info page and issue there.

agree!

**Comment by prettysync** [comment]
great idea!

**Reply by ntimasa** [reply]
Hi,
Thanks
Regards
Santho

**Comment by prettysync** [comment]
❤️

**Comment by mjovanc** [comment]
I like the idea! Sometimes you want to ask a question directly to a developer instead of writing an issue to get in contact. Sometimes email or other kinds of ways are not possible since they do not share it.

**Comment by rubaiyet-anjum** [comment]
This post on stackoverflow from 2012 says Github used to have a private messaging function. Wonder why they removed it.

https://stackoverflow.com/questions/12686545/how-to-leave-a-message-for-a-github-com-user

**Reply by MeesJ** [reply]
That's interesting!

**Reply by filipatbnp** [reply]
spamming?

**Comment by lukaseder** [comment]
I would be interested in this feature purely on a per-issue basis. E.g. if someone comments on an issue, I would like to be able to provide some additional information to that person in the context of the issue. It could go both ways, e.g. to post sensitive additional information. No need to contact that person outside of the issue (for me).

In other words, when an issue has a list of participants, it should be possible to add comments visible only to a selection of the participants (plus official team members of the project)

**Comment by AlexSkrypnyk** [comment]
The use case is - discussing sponsorship of a specific issue. People do not want to have money conversations in public.

**Comment by CatsMiaow** [comment]
If [GitHub Discussions](https://docs.github.com/en/discussions) adds the ability to create private discussion that can only be viewed by the owner or member of the repository, I think it could be a good replacement.

If so, you could create discussions in your [personal profile repository](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/customizing-your-profile/managing-your-profile-readme#adding-a-profile-readme) and use them for personal use.

**Comment by Tornadoalley** [comment]
I am shocked this still isn't a feature!

**Reply by ryszardep** [reply]
Same here, I wonder why this isn't up and running yet!

**Comment by ginjo** [comment]
I've had a need to contact someone privately (non-public) on Github a number of times over the years, for discussing issues that aren't the public's concern, or for discussing topics that are not specific to any one repository. And sometimes, I want others to be able to contact ME on Github – and I don't follow them or even know their accounts.

But most glaringly, there's no way to take a public conversation private between consenting members, without exposing sensitive information in a public forum.  Here's something I've wanted many times:

> Hey \[other user\], we've been wracking our brains on this issue for days. I'll bet we can clear this up in 10 minutes with a phone/zoom/skype call.  If interested, PM me and I'll send my digits.

I understand the potential for abuse with any kind of direct messaging, and the can of worms opened to manage such a feature, but there must be some way to minimize the downside with filters, permissions, or some kind of intelligent logic. Hoping this gets more attention.

**Comment by reivaxy** [comment]
one possible solution is to create a dummy private repository and add the person(s) you want to talk to as collaborators on it :)

**Comment by wilson0x4d** [comment]
Possibly unpopular opinion, but all of the privacy-related reasons cited by others are covered by using Slack, Skype, IRC, E-mail, Phone, CW, Smoke Signals, Two Cans connected by String, Rude Hand Gestures, etc.

It's as easy as developers publishing a public contact method on their profile, visible to other users. This allows Github to not waste money on something that is not their core-business (they spend money to manage our code and workflows, not social interactions and abuse.)

The simplest solution to the problem is to not create a problem in the first place. I am certain this is why Github removed the old messaging feature, it created a problem. The moment you make it possible for an abusive user to DM thousands of people (1) now I have to deal with that noise here too?! and (2) now github has to burn time and effort keeping it under control? Prefer they stay focused on their core business, I can already communicate freely elsewhere.

Don't hate.

**Reply by zufuliu** [reply]
SourceForge still has this feature:
<img width="617" height="384" alt="image" src="https://github.com/user-attachments/assets/a046921f-f051-4d0b-b288-d1afa968a057" />

**Comment by mbana** [comment]
I'd also like for this feature to be implemented.
